pkg/reconciler/topic/resources/publisher.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	servingv1beta1 "github.com/knative/serving/pkg/apis/serving/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/kmeta"
+
+	"github.com/GoogleCloudPlatform/cloud-run-events/pkg/apis/pubsub/v1alpha1"
+)
+
+// PublisherArgs are the arguments needed to create a Topic publisher.
+// Every field is required.
+type PublisherArgs struct {
+	Image  string
+	Topic  *v1alpha1.Topic
+	Labels map[string]string
+}
+
+const (
+	credsVolume    = "google-cloud-key"
+	credsMountPath = "/var/secrets/google"
+)
+
+// DefaultSecretSelector is the default secret selector used to load the creds
+// for the publisher to auth with Google Cloud.
+func DefaultSecretSelector() *corev1.SecretKeySelector {
+	return &corev1.SecretKeySelector{
+		LocalObjectReference: corev1.LocalObjectReference{
+			Name: "google-cloud-key",
+		},
+		Key: "key.json",
```

Comment:
This `key.json` is what I wanted to be documented earlier in the Topic type's Secret ref field. How does the user know to use this key otherwise?

Reply:
ack https://github.com/GoogleCloudPlatform/cloud-run-events/issues/64